### ISSUE TITLE
Asking for Image undistortion.

### ISSUE THREAD

**Issue by baopanda** [cvg/GeoCalib#18]
Hi authors.

![image](https://github.com/user-attachments/assets/913faf30-8ae9-41f8-953b-26b55e6f94c3)

Thank you for your insightful paper. 
I would like to inquire about how I can apply the methods to obtain the images mentioned above.
Could you please share which methods you used and how I can implement them? I have tried your calibration code, but it did not yield the results I was expecting. 

Thank you very much.

**Comment by veichta** [comment]
Hi,

The current method in the camera model undistorts the image without padding.
If you want the padded images as shown in the paper, you can use the function below and adjust the padding as needed:


```python
def undistort_image(img, camera, padding=0.3):
    """Undistort an image with optional padding.

    Args:
        img (numpy.ndarray): The image to undistort, with shape (H, W, 3).
        camera: The camera model.
        padding (float): The amount of padding to add (default: 0.3).

    Returns:
        numpy.ndarray: The undistorted image.
    """
    W, H = camera.size.unbind(-1)
    H, W = H.int().item(), W.int().item()

    pad_h, pad_w = int(H * padding), int(W * padding)
    x, y = torch.meshgrid(
        torch.arange(0, W + pad_w), torch.arange(0, H + pad_h), indexing="xy"
    )
    coords = torch.stack((x, y), dim=-1).reshape(-1, 2) - torch.tensor([pad_w / 2, pad_h / 2])

    p3d, _ = camera.pinhole().image2world(coords.to(camera.device).to(camera.dtype))
    p2d, _ = camera.world2image(p3d)

    p2d = p2d.float().numpy().reshape(H + pad_h, W + pad_w, 2)
    img = cv2.remap(img, p2d[..., 0], p2d[..., 1], cv2.INTER_LINEAR, borderValue=(0, 0, 0))

    return cv2.resize(img, (W, H))
```
